packages/core/scripts/hardhat/tasks/registry.js

```diff
@@ -0,0 +1,33 @@
+const { task } = require("hardhat/config");
```

Comment:
I generally find the task syntax to be more pleasant than the truffle syntax. Do you prefer this to a normal node script? What do you think the tradeoffs are between this and a normal node script for this sort of script?

Reply:
I agree! you also get CLI input args and params super easily along, and tasks can call each other easily as well: https://hardhat.org/guides/create-task.html

I think we should switch all of our scripts over to hardhat tasks because we can remove all of the logic that sets up parts of web3.

Reply:
I think the big downside is coupling ourselves (again) to a specific framework. I can see arguments either way. But definitely agreed. Might be nice to avoid web3/provider setup, although, we could always write a utility function to do all that work for us (or just use getWeb3 :)).